dsl/camel-yaml-dsl/camel-yaml-dsl/src/test/groovy/org/apache/camel/dsl/yaml/KameletTest.groovy

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.dsl.yaml
+
+import org.apache.camel.component.mock.MockEndpoint
+import org.apache.camel.dsl.yaml.common.YamlDeserializationMode
+import org.apache.camel.dsl.yaml.support.YamlTestSupport
+import org.apache.camel.processor.aggregate.UseLatestAggregationStrategy
+import org.apache.camel.spi.Resource
+
+class KameletTest extends YamlTestSupport {
+    @Override
+    def doSetup() {
+        context.start()
+    }
+
+    def "kamelet (#resource)"(Tuple2<YamlDeserializationMode, Resource> resource) {
+        setup:
+            addTemplate('setPayload') {
+                from('kamelet:source')
+                    .setBody().constant('{{payload}}')
+            }
+
+            setFlowMode(resource[0] as YamlDeserializationMode)
+            loadRoutes(resource[1] as Resource)
+
+            withMock('mock:kamelet') {
+                expectedMessageCount 1
+                expectedBodiesReceived '1'
+            }
+        when:
+            withTemplate {
+                to('direct:start').withBody('a').send()
+            }
+
+        then:
+            MockEndpoint.assertIsSatisfied(context)
+
+        where:
+            resource << [
+                new Tuple2<YamlDeserializationMode, Resource>(
+                    YamlDeserializationMode.CLASSIC,
+                    asResource('name', '''
+                        - from:
+                            uri: "direct:start"
+                            steps:
+                              - kamelet: 
+                                  name: "setPayload?payload=1"
+                                  steps:
+                                    - to: "mock:kamelet"
```

Comment:
@davsclaus it looks like I cannot use the shorter syntax here:

```yaml
- from:
    uri: "direct:start"
    steps:
      - kamelet: "setPayload?payload=1"
      - to: "mock:kamelet"
```

fails with 

```
org.apache.camel.FailedToCreateRouteException: Failed to create route route1 at: >>> Kamelet[[]] <<< in route: Route(route1)[From[direct:start] -> [Kamelet[[]], To[mock:ka... because of Definition has no children on Kamelet[[]]

	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:240)
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:74)
	at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49)
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:684)
	at org.apache.camel.impl.DefaultModel.addRouteDefinitions(DefaultModel.java:120)
	at org.apache.camel.impl.DefaultCamelContext.addRouteDefinitions(DefaultCamelContext.java:239)
	at org.apache.camel.builder.RouteBuilder.populateRoutes(RouteBuilder.java:569)
	at org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext(RouteBuilder.java:471)
	at org.apache.camel.impl.engine.AbstractCamelContext.addRoutes(AbstractCamelContext.java:1149)
	at org.apache.camel.spi.RoutesLoader.loadRoutes(RoutesLoader.java:43)
	at org.apache.camel.dsl.yaml.support.YamlTestSupport.loadRoutes(YamlTestSupport.groovy:60)
	at org.apache.camel.dsl.yaml.support.YamlTestSupport.loadRoutes(YamlTestSupport.groovy:75)
	at org.apache.camel.dsl.yaml.KameletTest.kamelet (#resource)(KameletTest.groovy:39)
Caused by: java.lang.IllegalArgumentException: Definition has no children on Kamelet[[]]
	at org.apache.camel.reifier.ProcessorReifier.createChildProcessor(ProcessorReifier.java:573)
	at org.apache.camel.component.kamelet.KameletReifier.createProcessor(KameletReifier.java:32)
	at org.apache.camel.component.kamelet.KameletProcessorFactory.doCreateProcessor(KameletProcessorFactory.java:32)
	at org.apache.camel.component.kamelet.KameletProcessorFactory.doCreateProcessor(KameletProcessorFactory.java:24)
	at org.apache.camel.support.TypedProcessorFactory.createProcessor(TypedProcessorFactory.java:44)
	at org.apache.camel.processor.DefaultProcessorFactory.createProcessor(DefaultProcessorFactory.java:92)
	at org.apache.camel.reifier.ProcessorReifier.makeProcessor(ProcessorReifier.java:833)
	at org.apache.camel.reifier.ProcessorReifier.addRoutes(ProcessorReifier.java:579)
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:236)
	... 12 more
```

Reply:
Okay fixed, try again with latest code

Reply:
it works now